src/sections/edit-dataset-terms/dataset-terms-tab/DatasetTermsTab.tsx

```diff
@@ -0,0 +1,294 @@
+import { useEffect, useMemo, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from 'react-router-dom'
+import { useForm, Controller, FormProvider } from 'react-hook-form'
+import { toast } from 'react-toastify'
+import { Form, Row, Col, Button, Alert } from '@iqss/dataverse-design-system'
+import { CustomTerms } from '../../../dataset/domain/models/Dataset'
+import { LicenseRepository } from '../../../licenses/domain/repositories/LicenseRepository'
+import { useGetLicenses } from './useGetLicenses'
+import { DatasetRepository } from '@/dataset/domain/repositories/DatasetRepository'
+import { useDataset } from '../../dataset/DatasetContext'
+import { useUpdateDatasetLicense } from './useUpdateDatasetLicense'
+import { Route, QueryParamKey } from '../../Route.enum'
+import {
+  DatasetNonNumericVersionSearchParam,
+  DatasetPublishingStatus
+} from '../../../dataset/domain/models/Dataset'
+import styles from './DatasetTermsTab.module.scss'
+
+interface DatasetTermsFormData {
+  license: string
+  customTerms?: CustomTerms
+}
+
+interface DatasetTermsTabProps {
+  licenseRepository: LicenseRepository
+  datasetRepository: DatasetRepository
+}
+
+export function DatasetTermsTab({ licenseRepository, datasetRepository }: DatasetTermsTabProps) {
+  const { t } = useTranslation('dataset')
+  const { t: tShared } = useTranslation('shared')
+  const { dataset, refreshDataset } = useDataset()
+  const navigate = useNavigate()
+
+  const formContainerRef = useRef<HTMLDivElement>(null)
+
+  const { licenses, isLoadingLicenses, errorLicenses } = useGetLicenses({
+    licenseRepository,
+    autoFetch: true
+  })
+
+  const { handleUpdateLicense, isLoading, error } = useUpdateDatasetLicense({
+    datasetRepository,
+    onSuccessfulUpdateLicense: () => {
+      toast.success(t('alerts.licenseUpdated.alertText'))
+      refreshDataset()
+    }
+  })
+
+  const initialCustomTerms = useMemo((): CustomTerms => {
+    if (dataset?.termsOfUse.customTerms) {
+      return dataset.termsOfUse.customTerms
+    }
+    return {
+      termsOfUse: '',
+      confidentialityDeclaration: '',
+      specialPermissions: '',
+      restrictions: '',
+      citationRequirements: '',
+      depositorRequirements: '',
+      conditions: '',
+      disclaimer: ''
+    }
+  }, [dataset?.termsOfUse.customTerms])
+
+  const licenseOptions = useMemo(() => {
+    const dynamicOptions = licenses
+      .sort((a, b) => a.sortOrder - b.sortOrder)
+      .map((license) => ({
+        value: license.id.toString(),
+        label: license.name,
+        uri: license.uri,
+        iconUri: license.iconUri || '',
+        isDefault: license.isDefault
+      }))
+
+    dynamicOptions.push({
+      value: 'CUSTOM',
```

Comment:
since this hard-coded 'CUSTOM' string is used in multiple places, would be good to replace it with a const.